usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
+            if (that.entityProperties.userData.length !== 0) {
+                try {
+                    that.userData = JSON.parse(that.entityProperties.userData);
+                } catch (e) {
+                    console.log(e, "no userData found");
+                    return;
+                }
+                that.token = that.userData.token;
+                that.expireTime = that.userData.expireTime;
+                that.sentAlready = false;
+                that.googleRequest(that.token);
+            }
+        }
+    };
+
+
+    this.googleRequest = function(token) {
+        var tomorrowMidnight = new Date();
+        tomorrowMidnight.setHours(HOURS_PER_DAY, 0, 0, 0);
+        that.scheduleURL = "https://www.googleapis.com/calendar/v3/calendars/" + 
+            that.calendarID + 
+            "/events?&timeMin=" + (new Date()).toISOString() +
+            "&timeMax=" + tomorrowMidnight.toISOString() +
+            "&showDeleted=" + false + 
+            "&singleEvents=" + true +
+            "&maxResults=2" +
+            "&orderBy=startTime" +
+            "&access_token=" + token;
+        if (!that.interval) {
+            that.interval = Script.setInterval(function() {
+                if ((new Date()).valueOf() > (that.expireTime - EXPIRY_BUFFER) && !that.sentAlready) {
+                    that.sentAlready = true;
+                    console.log("SENDING MESSAGE FOR NEW TOKEN");
+                    Messages.sendMessage(CHANNEL, JSON.stringify({
+                        type: "REFRESH TOKEN",
+                        token: that.token,
+                        uuid: that.roomScheduleID
+                    }));
+                } else if ((new Date()).valueOf() > (that.expireTime)) {
+                    console.log("EXCEEDED EXPIRATION OF TOKEN");
+                    Script.clearInterval(that.interval);
+                    that.interval = false;
+                    return;
+                }
+                that.request(that.scheduleURL, function (error, response) {
+                    if (error) {
+                        console.log("could not complete request", error);
+                        Script.clearInterval(that.interval);
+                        that.interval = false;
+                        return;
+                    } else {
+                        that.clearEventList(that.entityID);
+                        console.log(JSON.stringify(response));
+                        var events = response.items;
+                        if (events.length > 0) {
+                            for (var i = 0; i < events.length; i++) {
+                                var event = events[i];
+                                var start = event.start.dateTime;
+                                var end = event.end.dateTime;
+                                var summary = event.summary;
+                                if (!start) {
```

Comment:
why is this necessary?

Reply:
This is from the example google calendar script, I'm guessing it handles the case where somebody hasn't entered exact time information, say for an all day event.  It'd be interesting to see if this can actually handle something like that.